### DISCUSSION TITLE
Is there a way to limit the amount of scrollback lines fed to kitty-scrollback specifically?

### DISCUSSION THREAD

**Discussion by J-Kappes** [mikesmithgh/kitty-scrollback.nvim#83] [Q&A] [answered]
I like having the option to scroll back 10k lines in kitty itself or with `less`, but when I want to yank something it's usually pretty recent, and I get the impression the scrollback size contributes to slow startup.

**Comment by mikesmithgh** [comment, marked as answer]
Hey @J-Kappes, could you provide the output of `:KittyScrollbackCheckHealth`. I'd like to view some of those configs.

> Is there a way to limit the amount of scrollback lines fed to kitty-scrollback specifically?

Yes, I have been neglecting some documentation https://github.com/mikesmithgh/kitty-scrollback.nvim/issues/4 😹

My current and recommended `kitty.conf` settings are:


[scrollback_pager_history_size](https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.scrollback_pager_history_size) 2
- scrollback_pager_history_size controls the amount of history that is stored and passed to scrollback pager programs like kitty-scrollback.nvim or less. From the docs:
> The current implementation stores the data in UTF-8, so approximately 10000 lines per megabyte at 100 chars per line, for pure ASCII, unformatted text. 
- I pick 2 here because 2 mb is approximately 20000 lines (actually less than 20000 due to ANSI formatting) which seems like a reasonably large amount of history to me. 
[scrollback_fill_enlarged_window](https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.scrollback_fill_enlarged_window) yes

[scrollback_lines](https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.scrollback_lines) 2000
- scrollback_lines controls the number of lines visible in your Kitty terminal session (not a pager like less or kitty-scrollback.nvim). I just left as the default value 2000


Neovim allows for the maximum of 100000 lines in its terminal buffer (which kitty-scrollback.nvim uses). See [vim.o.scrollback](https://neovim.io/doc/user/options.html#'scrollback')

My guess is that you may have a large number for `scrollback_pager_history_size` which is causing the slow startup time for kitty-scrollback.nvim.

Hopefully this helps. I am interested in the results of `:KittyScrollbackCheckHealth` and if changing any of those values help. There is also a `show_timer` option in kitty-scrollback.nvim's config
```
        status_window = {
          show_timer = true,
        },
```
if you set that to true, it will show a timer while kitty is loading to get a rough number of how long it is taking.

**Reply by J-Kappes** [reply]
Here's my output before changing anything:
https://gist.github.com/J-Kappes/02fa67c371b7356efe4f2b97b0e44206

So if understand correctly, the only way of limiting `kitty-scrollback`'s buffer also limits `less`'s? That's too bad because I get no noticeable delay when starting `less` and I like using it to find stuff from far back.

**Reply by mikesmithgh** [reply]
well that is to limit to any scrollback pager, so yes less or kitty-scrollback.nvim. I am confused though, because your config has
`scrollback_pager_history_size = 0`
so I wouldn't expect you to have much in your pager scrollback at all?

I don't see any difference at least on my machine with less and kitty-scrollback.nvim for large scrollbacks.

What type of delay are we talking about ms or seconds? It might be helpful you could capture a video I may be missing something.

**Reply by J-Kappes** [reply]
ms. They're pretty noticable though, since the terminal goes blank for the duration. Here's a video of me starting `less` followed by `nvim`.

[Screencast from 2023-12-01 10-01-00.webm](https://github.com/mikesmithgh/kitty-scrollback.nvim/assets/87168492/16270190-ad9c-4da5-ae95-7ea351ef387b)

**Reply by mikesmithgh** [reply]
@J-Kappes thanks for the video, that helps! Yes, that is expected/current behavior at the moment.

I agree it would be nice to have it transition as quickly as less. I think I'd have to experiment with some caching or background process to get it to the same speed as less.
